clients/linux/src/ui/text_editor.rs

```diff
@@ -0,0 +1,135 @@
+use std::sync::mpsc;
+
+use gtk::glib;
+use gtk::prelude::*;
+use gtk::subclass::prelude::*;
+
+glib::wrapper! {
+    pub struct TextEditor(ObjectSubclass<imp::TextEditor>)
+        @extends gtk::Widget, gtk::Box,
+        @implements gtk::Accessible;
+}
+
+impl Default for TextEditor {
+    fn default() -> Self {
+        Self::new(lb::Uuid::parse_str("00000000-0000-0000-0000-000000000000").unwrap())
+    }
+}
```

Comment:
why is this needed?

Reply:
It's not. I think I was under the impression at one point that gtk widgets _had_ to implement default.